prime-angular-frontend/src/app/modules/enrolment/pages/regulatory/regulatory.component.ts

```diff
@@ -133,16 +166,49 @@ export class RegulatoryComponent extends BaseEnrolmentProfilePage implements OnI
     }
 
     // Replace previous values on deactivation so updates are discarded
-    const { certifications, deviceProviderIdentifier } = this.enrolmentService.enrolment;
-    this.formState.patchValue({ certifications, deviceProviderIdentifier });
+    const { certifications, enrolleeDeviceProviders } = this.enrolmentService.enrolment;
+    this.formState.patchValue({ certifications, enrolleeDeviceProviders });
   }
 
-  protected onSubmitFormIsValid() {
-    // Enrollees can not have certifications and jobs
-    this.removeOboSites();
-    // Remove remote access data when enrollee is no longer eligible, e.g., licence type changes
-    if (this.cannotRequestRemoteAccess) {
-      this.removeRemoteAccessData();
+  public onSubmit() {
+    if (this.formUtilsService.checkValidity(this.form)) {
+
+      if (this.isDeviceProvider && this.formState.deviceProviderId.value) {
+        let siteName = this.deviceProviderSite ?
+          this.deviceProviderSite.siteName : "Site not found.";
+        let siteAddress = this.deviceProviderSite ?
+          `${this.deviceProviderSite.siteAddress}, ${this.deviceProviderSite.city}, ${this.deviceProviderSite.prov}` : "";
+        let message = this.deviceProviderSite ?
+          "Is this the correct location where you are the device provider?" : "If you are not sure the Device Provider ID, you might continue."
```

Comment:
Would read better if "If you are not sure of the Device Provider ID, you may continue."

Reply:
Good call, updated.